c/aws-c-common/patch.diff

```diff
@@ -223,3 +223,79 @@ index 3e4e71b..f9e0d04 100644
  /* clang-format off */
  #    define AWS_IS_ZEROED(object)                                                                                      \
          __CPROVER_forall {                                                                                             \
+diff --git a/source/ring_buffer.c b/source/ring_buffer.c
+index ff649f2..a1a4ddc 100644
+--- a/source/ring_buffer.c
++++ b/source/ring_buffer.c
+@@ -20,7 +20,7 @@
+ #ifdef CBMC
+ #    define AWS_ATOMIC_LOAD_PTR(ring_buf, dest_ptr, atomic_ptr, memory_order)                                          \
+         dest_ptr = aws_atomic_load_ptr_explicit(atomic_ptr, memory_order);                                             \
+-        assert(__CPROVER_POINTER_OBJECT(dest_ptr) == __CPROVER_POINTER_OBJECT(ring_buf->allocation));                  \
++        assert((dest_ptr) == (ring_buf->allocation));                  \
```

Comment:
I am not entirely sure those ring-buffer changes will work? I naively thought this was memcpy-only, but change for the ring buffer here and below make me wonder whether that's still correct.

Reply:
Indeed your suspicions are true, they are not correct anymore. I'd say to remove the ring-buffer cases that make use of the function. Should I?

Reply:
The slightly less drastic measure would be removing those assertions, if that works, or else you will indeed need to remove the benchmarks.

Reply:
I tested all the original ring-buffer harnesses and the only one that fails without the `__CPROVER_POINTER_OBJECT` was `aws_ring_buffer_buf_belongs_to_pool`. Thus, I removed this test case, removed the assertion and updated the patch. Could you review again @tautschnig ?